### PR TITLE
Add support for screen-space contact shadows

### PR DESCRIPTION
**Pull request by pixelflinger**
Large scene using LiSPSM, without and with screen-space contact shadows:

without cs
with cs

### CODE REVIEW

File: shaders/src/light_directional.fs [hunk 1]
```diff
@@ -36,14 +36,44 @@ void evaluateDirectionalLight(const MaterialInputs material,
     Light light = getDirectionalLight();
 
     float visibility = 1.0;
+
 #if defined(HAS_SHADOWING)
     if (light.NoL > 0.0) {
+
+        // cast a ray in the direction of the light
```
Comment (bejado) [inline shaders/src/light_directional.fs]: This all should probably go within the `if(frameUniforms.directionalShadows)` block.

Comment (romainguy) [reply shaders/src/light_directional.fs]: We will most likely want a different uniform to control contact shadows.  That code should be into shadowing.fs though before the PR is ready for review.

Comment (pixelflinger) [reply shaders/src/light_directional.fs]: yes and yes

File: android/filament-android/src/main/java/com/google/android/filament/RenderableManager.java [hunk 2]
```diff
@@ -284,6 +284,18 @@ public Builder receiveShadows(boolean enabled) {
             return this;
         }
 
+        /**
+         * Controls if this renderable uses screen-space contact shadows. This is more
+         * expensive but can improve the quality of shadows, especially in large scenes.
+         * Ignored is this renderable is not a shadow caster.
```
Comment (bejado) [inline android/filament-android/src/main/java/com/google/android/filament/RenderableManager.java]:
```suggestion
         * Ignored if this renderable is not a shadow caster.

```

File: android/filament-android/src/main/java/com/google/android/filament/RenderableManager.java [hunk 3]
```diff
@@ -460,6 +472,15 @@ public void setReceiveShadows(@EntityInstance int i, boolean enabled) {
         nSetReceiveShadows(mNativeObject, i, enabled);
     }
 
+    /**
+     * Changes whether or not the renderable can uses screen-space contact shadows.
```
Comment (bejado) [inline android/filament-android/src/main/java/com/google/android/filament/RenderableManager.java]:
```suggestion
     * Changes whether or not the renderable can use screen-space contact shadows.

```

File: filament/include/filament/RenderableManager.h [hunk 4]
```diff
@@ -228,6 +228,14 @@ class UTILS_PUBLIC RenderableManager : public FilamentAPI {
          */
         Builder& receiveShadows(bool enable) noexcept;
 
+        /**
+         * Controls if this renderable uses screen-space contact shadows. This is more
+         * expensive but can improve the quality of shadows, especially in large scenes.
+         * Ignored is this renderable is not a shadow caster.
```
Comment (bejado) [inline filament/include/filament/RenderableManager.h]:
```suggestion
         * Ignored if this renderable is not a shadow caster.
```

File: filament/include/filament/RenderableManager.h [hunk 5]
```diff
@@ -343,6 +351,13 @@ class UTILS_PUBLIC RenderableManager : public FilamentAPI {
      */
     void setReceiveShadows(Instance instance, bool enable) noexcept;
 
+    /**
+     * Changes whether or not the renderable can uses screen-space contact shadows.
```
Comment (bejado) [inline filament/include/filament/RenderableManager.h]:
```suggestion
     * Changes whether or not the renderable can use screen-space contact shadows.
```

File: libs/filabridge/include/private/filament/SibGenerator.h [hunk 6]
```diff
@@ -42,8 +42,9 @@ struct PerViewSib {
     static constexpr size_t IBL_SPECULAR   = 4;
     static constexpr size_t SSAO           = 5;
     static constexpr size_t SSR            = 6;
+    static constexpr size_t STRUCTURE      = 7;
 
-    static constexpr size_t SAMPLER_COUNT = 7;
+    static constexpr size_t SAMPLER_COUNT = 8;
```
Comment (bejado) [inline libs/filabridge/include/private/filament/SibGenerator.h]:
```suggestion
    static constexpr size_t SAMPLER_COUNT  = 8;
```

**Pull request merged**